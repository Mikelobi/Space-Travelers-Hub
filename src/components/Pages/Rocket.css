.rocket-list {
  list-style-type: none;
}

.list-item {
  display: flex;
  padding: 1rem 0.5rem;
}

.list-item img {
  width: 200px;
  height: 150px;
}

.list-desc {
  text-align: left;
  padding: 0 1rem;
}

.list-desc h2 {
  padding: 0;
  margin: 0;
}

.reserve-button {
  border-radius: 4px;
  border: 0;
  background-color: #007BFF;
  color: #fff;
  padding: 0.5rem 1rem;
}

.cancel-button {
  border-radius: 4px;
  border: 1px solid #000;
  padding: 0.5rem 1rem;
}