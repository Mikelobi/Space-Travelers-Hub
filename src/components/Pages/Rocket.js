import React from 'react';
import { useSelector } from 'react-redux';
import ListItem from '../rockets/rocketList';

import './Rocket.css';

const Rocket = () => {
  // sample display
  const rocketList = useSelector((state) => state.rockets.list);
  const list = rocketList.map((item) => {
    const image = item.flickr_images[0];
    return (
      <ListItem
        key={item.id}
        id={item.id}
        name={item.name}
        description={item.description}
        image={image}
      />
    );
  });
  return <ul className="rocket-list">{list}</ul>;
};

export default Rocket;
