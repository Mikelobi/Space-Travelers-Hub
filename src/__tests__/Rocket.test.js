import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rocket from '../components/Pages/Rocket';


describe('Testing the rendering of rockets', () => {
  test('display empty list of rockets', () => {
    const { container } = render(
      <Provider store={store}>
        <Rocket />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
