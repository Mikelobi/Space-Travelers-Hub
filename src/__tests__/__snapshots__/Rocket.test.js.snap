// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the rendering of rockets display empty list of rockets 1`] = `
<div>
  <ul
    class="rocket-list"
  />
</div>
`;
